### PR TITLE
feat: In-extension dismissible upsells for Roo Code Cloud

### PR DESCRIPTION
**Pull request by brunobergher**
# Description

Adds in-app upsells for Roo Code Cloud.
Introduces a new DismissibleUpsell React component, which ensures messages are displayed in a consistent way and enforces dismissal logic, so we don't annoy users. The dismissal state is persisted in the VS Code extension's global state, ensuring they do not reappear after a user has closed them.

# Key implementation details include:

- A shared DismissibleUpsell component
- A centralized CloudUpsellDialog component, which is called from different places
- Integration of these upsell CTAs into several key user workflows:
  - On the extion home page.
  - Within the "AutoApprove" settings menu.
  - After a long-running task runs for a while.
- CloudUpsellDialog is also used in the "Share Task" and "Cloud View" screens, reducing duplication
- Robust state management to handle component mounting/unmounting and prevent common issues like race conditions or memory leaks during asynchronous operations.

# Test Procedure
The functionality was tested via a combination of new unit tests and manual verification.

## Unit Tests:
Comprehensive tests were added for the DismissibleUpsell component, covering its variants, dismissal logic, and various edge cases such as rapid dismissals and state updates on unmounted components.
Tests for the auto-approve upsell functionality were also created and fixed.

## Manual Testing:
To verify the changes, reviewers can perform the following steps:

1. Navigate to the Roo Code home page and confirm the new CTA is visible.
2. Open the "AutoApprove" menu and enable several options
3. Run a task that takes a significant amount of time and check that the upsell appears after a while
4. Dismiss one of the upsells, then restart the extension to ensure it remains dismissed.
5. Verify that all new UI text has been correctly internationalized.

# Screenshots

----

> [!IMPORTANT]
> Introduces in-app dismissible upsells for Roo Code Cloud with new React components, state management, and internationalization support.
> 
>   - **Behavior**:
>     - Adds `DismissibleUpsell` React component for consistent message display and dismissal logic.
>     - Integrates upsells into workflows: extension home page, "AutoApprove" settings, long-running tasks, "Share Task", and "Cloud View".
>     - Uses `CloudUpsellDialog` to reduce duplication in "Share Task" and "Cloud View".
>     - Persists dismissal state in VS Code extension's global state.
>   - **State Management**:
>     - Manages component mounting/unmounting to prevent race conditions and memory leaks.
>   - **Testing**:
>     - Adds unit tests for `DismissibleUpsell` covering variants, dismissal logic, and edge cases.
>     - Tests auto-approve upsell functionality.
>     - Manual testing includes verifying CTA visibility, dismissal persistence, and internationalization.
>   - **Internationalization**:
>     - Updates translations in multiple languages to support new upsell messages.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=RooCodeInc%2FRoo-Code&utm_source=github&utm_medium=referral)<sup> for 9610626ee97b5401099f00c93a8bb9e301270b36. You can [customize](https://app.ellipsis.dev/RooCodeInc/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: .roo/rules-merge-resolver/1_workflow.xml [hunk 1]
```diff
@@ -30,12 +30,13 @@
       <tools>
         <tool>gh pr checkout [PR_NUMBER] --force</tool>
         <tool>git fetch origin main</tool>
-        <tool>git rebase origin/main</tool>
+        <tool>GIT_EDITOR=true git rebase origin/main</tool>
```
[the remark this answers is not in the document]
Comment (brunobergher) [reply .roo/rules-merge-resolver/1_workflow.xml]: I have no idea how they ended up here. If a reviewer has any thoughts, I'm all ears.

File: webview-ui/src/components/common/DismissibleUpsell.tsx [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	variant?: "default" | "banner"
+	/** Optional callback when upsell is dismissed */
+	onDismiss?: () => void
+	/** Optional callback when upsell is clicked */
+	onClick?: () => void
+	/** Whether clicking the upsell should also dismiss it (default: false) */
+	dismissOnClick?: boolean
+}
+
+const DismissIcon = () => (
```
Comment (daniel-lxs) [inline webview-ui/src/components/common/DismissibleUpsell.tsx:24]: Does it make sense to use Lucide icons here?
